test/models/location_test.rb

```diff
@@ -1,6 +1,62 @@
 require 'test_helper'
 require 'models/shared/taxonomies_base_test'
+require 'rfauxfactory'
 
 class LocationTest < ActiveSupport::TestCase
   include TaxonomiesBaseTest
+  valid_loc_data_list = [
+    RFauxFactory.gen_alpha(1),
+    RFauxFactory.gen_alpha(246),
+    *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,
+    RFauxFactory.gen_html(rand((1..221)))
+  ]
+  invalid_loc_name_list = [
+    '',
+    ' ',
+    "\t",
+    *RFauxFactory.gen_strings(247).values
+  ]
+
+  test 'should create with multiple valid names' do
+    valid_loc_data_list.each do |name|
+      location = FactoryBot.create(:location, :name => name)
+      location.reload
+      assert_equal name, location.name
+    end
+  end
+
+  test 'should not create with multiple invalid names' do
+    invalid_loc_name_list.each do |name|
+      location = Location.new(:name => name)
+      refute location.valid?
+    end
+  end
+
+  test 'should update with multiple valid names' do
+    location = FactoryBot.create(:location, :name => name)
+    valid_loc_data_list.each do |name|
+      location.name = name
+      location.save
+      updated_location = Location.find_by_id(location.id)
+      assert_equal name, updated_location.name
+    end
+  end
+
+  test 'should update with multiple valid descriptions' do
+    location = FactoryBot.create(:location)
+    valid_loc_data_list.each do |description|
```

Comment:
it seems you rely on fact that the same data is valid for name and description which might not necessarily be true. While name is string with limitation of 255 chars, description is long text. I think this test could be simplified and only verify one valud for description that you generate here, ideally longer than 255 chars.